### ISSUE TITLE
Allow better control over sudo() exec environment/flags

### ISSUE THREAD

**Issue by bitprophet** [fabric/fabric#503]
Two strongly connected use cases re: env:
- User invokes `sudo` and expects it to preserve the invoking/connecting user's environment (e.g. `SSH_AUTH_SOCK` when tying it in with #72 and, say, trying to `sudo git clone`.)
  - This requires that the user being sudo'd as has `ALL` or `SETENV` set in their privilege spec, and that sudo is invoked as `sudo -E`.
- User invokes `sudo(xxx, user=foo)` and expects it to act with `foo`'s login environment.
  - This requires invoking as `sudo -i` and may also be subject to the target's sudoers config.

While some sudoers configs will make it impossible, I believe that the most common defaults + us adding additional `sudo` CLI flags to our invoke may make this less painful for users.

There are a number of other situations that also require sudo flags, such as `-k/-K`, `-H`, and so on. These will all need to be solved the same way.

Basically, we should either add some human-readable options to `sudo()` for common flags, or allow custom flag control (and/or expose `env.sudo_prefix` to users, as seen in #432.)

**Comment by liammcdermott** [comment]
My use-case for `-H` is that `-i` demands a password, this is no good for an unattended task (where a user has restricted access to certain commands without entering a password).

I may be doing it wrong, of course.

**Comment by elgow** [comment]
The workaround from above 
```python
with settings(sudo_prefix='sudo -i -p \'{}\''.format(env.sudo_prompt)):
```
is not really satisfactory. It requires understanding the internals of Fabric and is far too easily broken by messing up the sudo_prompt portion. It would be much better to provide a simple way to get the -H option because important utilities such as pip use ~ under the covers in their operation. I'd even suggest that setting the home dir to that of the target user, at least when that user is not root, should be the default.
